client.go

```diff
@@ -115,9 +124,34 @@ func (c *Client) request(ctx context.Context, method string, urlStr string, quer
 	req.Header.Add("Notion-Version", c.notionVersion)
 	req.Header.Add("Content-Type", "application/json")
 
-	res, err := c.httpClient.Do(req.WithContext(ctx))
-	if err != nil {
-		return nil, err
+	failedAttempts := 0
+	var res *http.Response
+	for {
+		var err error
+		res, err = c.httpClient.Do(req.WithContext(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		if res.StatusCode == http.StatusTooManyRequests {
+			failedAttempts++
+			if failedAttempts == 3 {
```

Comment:
I think this value shouldn't be hardcoded. It should be a part of the `Client` structure and be modified through `ClientOption`